### PR TITLE
doc: Scenario guide for VMware HTTP API usage

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY
This adds scenario guide for using VMware HTTP API using uri module.

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME
docs/docsite/rst/vmware/scenario_vmware_http.rst
docs/docsite/rst/vmware/vmware_scenarios.rst

##### ANSIBLE VERSION

```paste below
2.8-devel
```

### CODE REVIEW

File: docs/docsite/rst/vmware/scenario_vmware_http.rst [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+- All variable names and VMware object names are case sensitive.
+- You need to use Python 2.7.9 version in order to use ``validate_certs`` option, as this version is capable of changing the SSL verification behaviours.
+- VMware HTTP APIs are introduced in vSphere 6.5 and above so minimum level required in 6.5.
+- There are very limited number of APIs exposed, so you may need to rely on XMLRPC based VMware modules.
+
+
+Example Description
+===================
+
+With the following Ansible playbook you can find the VMware ESXi host system(s) and can perform various tasks depending on the list of host system.
```
Comment (samccann) [inline docs/docsite/rst/vmware/scenario_vmware_http.rst]:
```suggestion
With the following Ansible playbook you can find the VMware ESXi host system(s) and can perform various tasks depending on the list of host systems.
```

File: docs/docsite/rst/vmware/scenario_vmware_http.rst [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+- Hostname of the vCenter server
+- Username and password for the vCenter server
+- Version of vCenter is at least 6.5
+
+For now, you will be entering these directly, but in a more advanced playbook this can be abstracted out and stored in a more secure fashion using :ref:`ansible-vault` or using `Ansible Tower credentials <https://docs.ansible.com/ansible-tower/latest/html/userguide/credentials.html>`_.
+
+If your vCenter server is not setup with proper CA certificates that can be verified from the Ansible server, then it is necessary to disable validation of these certificates by using the ``validate_certs`` parameter. To do this you need to set ``validate_certs=False`` in your playbook.
+
+As you can see, we are using ``uri`` module in first task to login into the vCenter server and storing result in ``login`` variable using register. In second task, using cookies from first task we gathering information about ESXi host system.
```
Comment (samccann) [inline docs/docsite/rst/vmware/scenario_vmware_http.rst]:
```suggestion
As you can see, we are using the ``uri`` module in first task to login into the vCenter server and storing result in the ``login`` variable using register. In the second task, using cookies from the first task we are gathering information about the ESXi host system.
```

File: docs/docsite/rst/vmware/scenario_vmware_http.rst [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+
+Using this information, we are changing ESXi host system's advance configuration.
```
Comment (samccann) [inline docs/docsite/rst/vmware/scenario_vmware_http.rst]:
```suggestion
Using this information, we are changing the ESXi host system's advance configuration.
```

File: docs/docsite/rst/vmware/scenario_vmware_http.rst (hunk 1 above)
Comment (Akasurde) [reply docs/docsite/rst/vmware/scenario_vmware_http.rst]: Done

File: docs/docsite/rst/vmware/scenario_vmware_http.rst [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+    * We recommend installing the latest version with pip: ``pip install Pyvmomi`` on the Ansible control node
+      (as the OS packages are usually out of date and incompatible) if you are planning to use any existing VMware modules.
+
+* Hardware
+
+    * vCenter Server 6.5 and above with at least one ESXi server
+
+* Access / Credentials
+
+    * Ansible (or the target server) must have network access to the either vCenter server or the ESXi server
```
Comment (samccann) [inline docs/docsite/rst/vmware/scenario_vmware_http.rst]:
```suggestion
    * Ansible (or the target server) must have network access to either the vCenter server or the ESXi server
```

Comment (samccann) [reply docs/docsite/rst/vmware/scenario_vmware_http.rst]: sorry, found one last issue...

Comment (Akasurde) [reply docs/docsite/rst/vmware/scenario_vmware_http.rst]: np, Done.

**Review by samccann — APPROVED** [review]

**Pull request merged**